go-controller/pkg/network-controller-manager/network_controller_manager.go

```diff
@@ -340,6 +340,10 @@ func (cm *networkControllerManager) Start(ctx context.Context) error {
 			klog.Errorf("Unable to get nodes from informer while waiting for node zone sync")
 			return false, nil
 		}
+		if len(nodes) == 0 {
```

Comment:
since watch factory starts before we are confident nodes should be in the cache by then right?

Reply:
> since watch factory starts before we are confident nodes should be in the cache by then right?

yes. I am building on the assumptions taken when https://github.com/ovn-org/ovn-kubernetes/commit/5e6787d187a77d837b7bddade57e93c6fb199116 was implemented. If list is empty and there we no errors then I'm trusting the list is accurate.